setup.py

```diff
@@ -21,5 +21,6 @@ def test_reqs():
 
     packages=setuptools.find_packages(exclude=('tests')),
 
+    install_requires=['requests'],
```

Comment:
We should load the contents of the `requirements.txt` file, `install_requires` all the listed requirements; see how `test_reqs` does it.

Reply:
After reading https://packaging.python.org/discussions/install-requires-vs-requirements/ I understand that requirements.txt and install_requires are for different things. `requirements.txt` is for the "concrete" requirements, while `install_requires` is for the "abstract" requirements. The primary difference seems to be that one specifies exact versions to pin to, while the other provides more general restrictions.

The others reason why I didn't do it the same way was because even without the version numbers, the lists are different. The only package we require to run is `requests`, yet `requirements.txt` contains entries for all of `requests` dependencies too.

What do you think?

Reply:
Oh, I wasn't aware of the difference at all. Is there a minimal version of `requests` we need?

I think we have a bit of another anti parttern in our other python codebases, that essentially use the `requirements.txt` as a way to fill in the `install_requires`, which is overly specific.

I'm happy with the way it is as the moment, then. I think the only issue is that we have two sets of requirements that can go out of sync, but the solution is to update the requirements.txt from a fresh venv. I think this should be part of a build or release process.

Something like the following (untested).
```
set -e
virtualenv dist
. dist/bin/activate
pip install .
python setup.py test
pip freeze | grep -v pytest > requirements.txt
git commit -sm "[DOC] Update requirements" requirements.txt
```

... Did we have a discussion about `pip freeze` last sprint? My memory fails me.